index.js

```diff
@@ -10,25 +10,23 @@ function generate(argIndex) {
 		var promise = thunk(value);
 		thunk.then = promise.then.bind(promise);
 		thunk.catch = promise.catch.bind(promise);
-		return thunk;
 
-		function thunk(result) {
-			if (promise) {
-				// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
-				promise.catch(function () {});
-			}
+		// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
+		promise.catch(function () {});
 
-			return new Promise(function () {
-				// resolve / reject
-				var complete = arguments[argIndex];
+		return thunk;
 
-				setTimeout(function () {
-					complete(useValue ? value : result);
-				}, ms);
+		function thunk(result) {
+			return new Promise(function (resolve, reject) {
+				setTimeout(
+					willResolve ? resolve : reject,
+					ms,
+					useValue ? value : result
```

Comment:
Except we use this all over our AVA tests, and when we add browser support their, we are going to start caring.